pkg/dockerregistry/client.go

```diff
@@ -311,6 +318,11 @@ func (c *connection) checkV2() (bool, error) {
 		// handle auth challenges on individual repositories
 	case code >= 300 || resp.StatusCode < 200:
 		return false, nil
+	default:
+		if mediatype, _, err := mime.ParseMediaType(resp.Header.Get("Content-Type")); err == nil && mediatype != "application/json" {
```

Comment:
The *spec* doesn't say that there is any content.  Unfortunately, quay.io violates the spec.  

Actually, going to make this docker distribution header.  One sec.